src/cmdlib.sh

```diff
@@ -318,8 +321,19 @@ openshift_git_hack() {
     info "NOTICE: using workaround for Openshift missing .git"
 
     if [ "${GIT_URL:-x}" == "x" ] ||  [ "${GIT_REF:-x}" ]; then
-        info "Checking out bare git of ${GIT_URL} to ${gitd}/.git, ref ${GIT_REF}"
-        git clone --depth 1 --bare -b "${GIT_REF}" "${GIT_URL}" "${gitd}/.git"
+        tmpgit="$(mktemp -d)"
+        trap 'rm -rf ${tmpgit}; unset tmpgit;' EXIT
+
+        info "Re-creating git tree in ${gitd} from ${GIT_URL}, ref ${GIT_REF}"
+        git clone --depth 1 -b "${GIT_REF}" "${GIT_URL}" "${tmpgit}"/clone
+        rsync -av --ignore-existing "${tmpgit}"/clone/ "${gitd}"/
```

Comment:
I'm not a fan of this, but short of doing a full checkout, and bringing the changes back into the working tree, `git submodule` commands fail on EL7. The `git submodule` command in EL7 does not understand the "work tree" options that make this unneeded.